packages/next/build/webpack/loaders/next-flight-client-loader/index.ts

```diff
@@ -5,31 +5,50 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+import path from 'path'
 import { checkExports } from '../../../analysis/get-page-static-info'
 import { parse } from '../../../swc'
 
+function containsPath(parent: string, child: string) {
+  const relation = path.relative(parent, child)
+  return !!relation && !relation.startsWith('..') && !path.isAbsolute(relation)
+}
+
 export default async function transformSource(
   this: any,
   source: string
 ): Promise<string> {
-  const { resourcePath } = this
-
-  const transformedSource = source
-  if (typeof transformedSource !== 'string') {
+  if (typeof source !== 'string') {
     throw new Error('Expected source to have been transformed to a string.')
   }
 
-  const swcAST = await parse(transformedSource, {
-    filename: resourcePath,
-    isModule: 'unknown',
-  })
-  const { ssg, ssr } = checkExports(swcAST)
+  const appDir = path.join(this.rootContext, 'app')
+  const isUnderAppDir = containsPath(appDir, this.resourcePath)
+  const filename = path.basename(this.resourcePath)
+  const isPageOrLayoutFile = /^(page|layout)\.client\.\w+$/.test(filename)
+
+  const createError = (name: string) =>
+    new Error(
```

Comment:
If we can get the specific line number `getStaticProps`/`getServerSideProps` is at it would be nice if we could update the stack for the Error created here so the dev overlay can show the file/code